### PR TITLE
Hunter - fix back buttons (Issue #27)

### PR DESCRIPTION
**Pull request by Hunter27E**

### CODE REVIEW

**Review by k9kathyli — APPROVED** [review]

**Comment by Hunter27E** [comment]
Screen Shot 2022-05-16 at 6 26 08 PM

File: source/tests/PuppTests.test.js [hunk 1]
```diff
@@ -150,7 +150,7 @@ describe('basic navigation for BJ', () => {
         });
         await page.waitForTimeout(300);
 
-        page.on('dialog', async (dialog) => {});
+        // page.on('dialog', async (dialog) => {});
```
Comment (k9kathyli) [inline source/tests/PuppTests.test.js:153]: nit: remove comment?

File: source/YearlyOverview/YearlyOverview.html [hunk 2]
```diff
@@ -11,15 +11,15 @@
         <div id="header">
             <h1 id="index">
                 <a
-                    href="http://127.0.0.1:5500/source/Index/Index.html"
+                    href="../Index/Index.html"
                     style="text-decoration: none"
                     id="back"
                 >
-                    < Index</a
+                    Index</a
```
Comment (k9kathyli) [inline source/YearlyOverview/YearlyOverview.html:18]: Was the "<" removed on purpose? Stylistically, I liked it more

**Review by k9kathyli — COMMENTED** [review]
- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [x] Adequate test coverage
- [x] Follows coding conventions 
- [x] Code is commented/documented well

File: source/YearlyOverview/YearlyOverview.html (hunk 2 above)
Comment (Hunter27E) [reply source/YearlyOverview/YearlyOverview.html:18]: Yeah, the reason I moved it was cause the previous page wasn't actually Index. The appearance contradicts the functionality.

**Review by Dsuzuki126 — CHANGES_REQUESTED** [review]
- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [ ] UI looks as intended and fits guidelines
- [x] Adequate test coverage
- [x] Follows coding conventions
- [x] Code is commented/documented well

Could you also when you are on the yearly overview, make the index link in the upper left corner underline when you hover over it?
Thanks Hunter

File: source/MonthlyOverview/MonthlyStyles.css [hunk 3]
```diff
@@ -54,6 +54,10 @@ body {
     font-size: 3.5vh;
 }
 
+#back:hover {
+    text-decoration: underline !important;
```
[the remark this answers is not in the document]
Comment (Hunter27E) [reply source/MonthlyOverview/MonthlyStyles.css]: There was an inline style overriding the external stylesheet - it is resolved now.

File: source/DailyOverview/DailyStyles.css [hunk 4]
```diff
@@ -50,6 +50,14 @@ h2 {
     display: block;
 }
 
+h1 > a:first-child {
+    text-decoration: none;
+}
+
+h1 > a:first-child:hover {
+    text-decoration: underline !important;
```
[the remark this answers is not in the document]
Comment (Hunter27E) [reply source/DailyOverview/DailyStyles.css]: Resolved.

File: source/tests/PuppTests.test.js (hunk 1 above)
Comment (Hunter27E) [reply source/tests/PuppTests.test.js]: Resolved.

**Comment by Hunter27E** [comment]
Updated PR:
• removed !important
• updated Index link on YearlyOverview page to have hover state consistent with other links

**Review by Dsuzuki126 — CHANGES_REQUESTED** [review]
I think everything you did is good, the only changes I would suggest you make is fix the test messages for login tests in the PuppTests.test.js file lines 73, 89, 104, 120, and 136. If you look at the Login.js file that Sasha did you'll see the different responses that he made.  When you change the messages try running the tests. I believe they may fail as I made some changes to the tests on my PR so they work.

**Review by Dsuzuki126 — APPROVED** [review]
The tests need to be fixed so will need to merge the test fix branch after making sure it is up to date with the main branch

**Review by k9kathyli — APPROVED** [review]

**Pull request merged**